includes/class-sensei-modules.php

```diff
@@ -205,34 +207,57 @@ public function get_lesson_module_if_exists( $post ) {
 		return $lesson_module;
 	}
 
-	private function render_module_select_for_course( $lesson_course, $lesson_module = 0 ) {
-		// Get the available modules for this lesson's course
-		$modules = $this->get_course_modules( $lesson_course );
+	/**
+	 * Renders the lesson module select input.
+	 *
+	 * @param int|null $course_id
+	 * @param int|null $current_module_id
+	 *
+	 * @return string
+	 */
+	private function render_lesson_module_select_for_course( int $course_id = null, int $current_module_id = null ) {
+		// Get the available modules for this lesson's course.
+		$modules = $course_id ? $this->get_course_modules( $course_id ) : [];
+
+		// Build the HTML.
+		$input_name = 'lesson_module';
 
 		$html  = '';
 		$html .= '<input type="hidden" name="' . esc_attr( 'woo_lesson_' . $this->taxonomy . '_nonce' ) . '" id="' . esc_attr( 'woo_lesson_' . $this->taxonomy . '_nonce' ) . '" value="' . esc_attr( wp_create_nonce( plugin_basename( $this->file ) ) ) . '" />';
 
-		// Build the HTML to output
-		if ( is_array( $modules ) && count( $modules ) > 0 ) {
-			$html .= '<select id="lesson-module-options" name="lesson_module" class="widefat" style="width: 100%">' . "\n";
+		if ( $modules ) {
+			$html .= '<select id="lesson-module-options" name="' . esc_attr( $input_name ) . '" class="widefat" style="width: 100%">' . "\n";
 			$html .= '<option value="">' . esc_html__( 'None', 'sensei-lms' ) . '</option>';
 			foreach ( $modules as $module ) {
-				$html .= '<option value="' . esc_attr( absint( $module->term_id ) ) . '"' . selected( $module->term_id, $lesson_module, false ) . '>' . esc_html( $module->name ) . '</option>' . "\n";
+				$html .= '<option value="' . esc_attr( absint( $module->term_id ) ) . '"' . selected( $module->term_id, $current_module_id, false ) . '>' . esc_html( $module->name ) . '</option>' . "\n";
 			}
 			$html .= '</select>' . "\n";
 		} else {
-			$course_url = admin_url( 'post.php?post=' . urlencode( $lesson_course ) . '&action=edit' );
-
-			/*
-			 * translators: The placeholders are as follows:
-			 *
-			 * %1$s - <em>
-			 * %2$s - </em>
-			 * %3$s - Opening <a> tag to link to the Course URL.
-			 * %4$s - </a>
-			 */
-			$html .= '<p>' . wp_kses_post( sprintf( __( 'No modules are available for this lesson yet. %1$sPlease add some to %3$sthe course%4$s.%2$s', 'sensei-lms' ), '<em>', '</em>', '<a href="' . esc_url( $course_url ) . '">', '</a>' ) ) . '</p>';
+			$html .= '<input type="hidden" name="' . esc_attr( $input_name ) . '" value="">';
+
+			if ( $course_id ) {
+				$course_url = admin_url( 'post.php?post=' . $course_id . '&action=edit' );
+
+				/*
+				 * translators: The placeholders are as follows:
+				 *
+				 * %1$s - <em>
```

Comment:
It's actually considered to be better to include HTML directly in the string. For context, please see this [excellent resource](https://developer.wordpress.com/2015/04/23/wordpress-developers-test-your-i18n-internationalization-knowledge/) by an A11n, namely the second example. It seems like a good opportunity to correct it.

Reply:
Thanks for the info Donna, this localization wasn't done by me. I've just restructured the code. If I change it, that might break some of the translations. Should I still do it?

Reply:
That's the question of the day, isn't it? 😆 It's definitely a trade-off and I'm not sure what the right answer is. On the one hand, because it's in the backend it seems like less of a big deal to change than it would be if it were something that appeared on the frontend. On the other hand, is it _really_ that big a deal? Probably not. I'm good with leaving it. 👍🏻